freestyle/shared/src/main/scala/freestyle/free.scala

```diff
@@ -4,6 +4,10 @@ import scala.annotation.{compileTimeOnly, StaticAnnotation}
 import scala.language.experimental.macros
 import scala.reflect.macros.blackbox
 
+trait FreeModuleLike {
```

Comment:
This is also necessary for the IntelliJ integration and instead of calling it `FreeModuleLike` I think we should call it something like `Algebra`.  Unless the IDEA people get the `Inject` api they provide for libs that use macros fixed as described in #58 users will end up extending this trait to get proper syntax highlighting in IDEA like so:
```scala
@free trait MyOps[F[_]] extends Algebra {
   ....
}

MyOps.Op //visible by IntelliJ syntax highlighter when directly referenced and in `Handler` implementations.
```